prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -347,6 +350,54 @@ instance Pretty Char where
     prettyList = vsep . map unsafeTextWithoutNewlines . T.splitOn "\n"
 #endif
 
+-- | This class is similar to 'Pretty', but allows you to embed annotations in
+-- the 'Doc'.
+--
+-- @since 1.7.1
+class PrettyAnn ann a where
```

Comment:
What's the reason for the swapped order of type parameters compared to the proposal in #222?

Reply:
IIRC I wanted the `a` to be the last type parameter since it is the last type parameter for `Pretty` too; Hence given a fixed annotation type the signatures align:
```haskell
Pretty :: Type -> Constraint
PrettyAnn MyAnnotation :: Type -> Constraint
```
Then it is also easier to define aliases like
```haskell
type MyPretty = PrettyAnn MyAnnotation
```

Not a very strong argument, I know; Should I change that too?

Reply:
No, it's fine. I just saw that @andreasabel argued in favor of this order of parameters too: https://github.com/quchen/prettyprinter/pull/256#discussion_r1878633783